File: Dados Utilizados/Algoritmos e Listas/scripts-ordenados/java/ShellOrdenado.java
package java.ordenado;

import java.io.FileNotFoundException;

/**
 * Classe dedicada à avaliação de tempo do método de ordenação ShellSort.
 * A classe utilizará um vetor de números já ordenados.
 * @author Uriel Cairê
 */
public class ShellOrdenado {

    /**
     * No main é criado o vetor a ser utilizado;
     * É feita a chamada do ordenador por 30 vezes;
     * Por fim, os tempos de execução são exibidos em nanossegundos.
     * @param args the command line arguments
     */
    public static void main(String[] args) throws FileNotFoundException {
        int x = 20000;
        int array[] = new int[x];
		
		for (int i = 0; i < x; i++) {
            array[i] = i;
        }
        
        int[] arrayClone;
        
        for (int j = 0; j < 30; j++) {
            arrayClone = array.clone();
            long TempoInicial = System.nanoTime(); 
            ShellSort(arrayClone);
            long TempoTotal= System.nanoTime() - TempoInicial;
            System.out.println(TempoTotal);
        }
        
        
    }
    
    /**
     * Método de ordenação ShellSort.
     * @param vetor contém o um vetor de inteiros que devem ser ordenados.
     */
    public static void ShellSort(int[] vetor) {
        int h = 1;
        int n = vetor.length;
        while(h < n)
                h = h * 3 + 1;
        h = h / 3;
        int c, j;
        while (h > 0) {
            for (int i = h; i < n; i++) {
                c = vetor[i];
                j = i;
                while (j >= h && vetor[j - h] > c) {
                    vetor[j] = vetor[j - h];
                    j = j - h;
                }
                vetor[j] = c;
            }
            h = h / 2;
        }
    }
    
    
    
}

File: Dados Utilizados/Algoritmos e Listas/scripts-ordenados/python/ShellOrdenado.py
"""
Classe dedicada à avaliação de tempo do método de ordenação ShellSort.
O script utilizará utilizará um vetor de números já ordenados.
"""

"""
Método de ordenação ShellSort
O método recebe um array como parametro.
"""

#######shellsort########
def shellSort(array):
     gap = len(array) // 2
     # loop over the gaps
     while gap > 0:
         # do the insertion sort
         for i in range(gap, len(array)):
             val = array[i]
             j = i
             while j >= gap and array[j - gap] > val:
                 array[j] = array[j - gap]
                 j -= gap
             array[j] = val
         gap //= 2


lista = []
for i in range(20000):
    lista.append(i)

import timeit

arquivo = open('numeros.txt', 'w')
for i in range(30):
    inicio = timeit.default_timer()
    shellSort(lista)
    fim = timeit.default_timer()
    arquivo.write(str(fim-inicio))
    arquivo.write('\n')
arquivo.close()